### DISCUSSION TITLE
oc_ prefix on database username (custom install)

### DISCUSSION THREAD

**Discussion by Proxymiity** [nextcloud/all-in-one#3089] [Questions] [answered]
Hi,

I'm using a manual Nextcloud AIO install, using the provided .env file as well as the docker compose chart.
It sounds like the nextcloud container ignores its config.php and uses the .env (which is fine), but prepends a `oc_` prefix on the database's username.

I've been trying to debug this for 2 days now and cannot find out why.

The nextcloud container returns `psql: error: connection to server at "<redacted>" (<redacted>), port 5432 failed: FATAL:  password authentication failed for user "oc_hello_nextcloud"` while my .env clearly states `DATABASE_USER=hello_nextcloud`.

Has anyone got an idea how to remove the `oc_` prefixing?

**Comment by Zoey2936** [comment]
it can't be removed

**Reply by Proxymiity** [reply]
Why so? Is there a specific reason behind that?

**Reply by Zoey2936** [reply]
see here: https://github.com/nextcloud/all-in-one/blob/main/Containers/postgresql/init-user-db.sh

**Reply by Proxymiity** [reply]
Oh, okay. So, the way to get around this would be to build the container myself, right?

**Reply by Zoey2936** [reply]
No. Just don't set an env which is not supported and breaks AIO

**Reply by Zoey2936** [reply]
what .env file do you even use? https://github.com/nextcloud/all-in-one/blob/main/manual-install/sample.conf this env file doesn't list such and option, same as the Readme in the same folder

**Reply by Proxymiity** [reply]
I modified the docker-compose file to use an external database, so the .env adds a reference to the following: https://github.com/nextcloud/all-in-one/blob/main/manual-install/latest.yml#L80

**Reply by Zoey2936** [reply]
that is not supported, please follow the Readme

**Reply by Proxymiity** [reply]
Yeah, but the readme doesn't explain how to use an external database.

**Reply by mben42** [reply]
> that is not supported, please follow the Readme

That's stupid, please follow your users needs.

**Reply by Zoey2936** [reply]
> Yeah, but the readme doesn't explain how to use an external database.

you don't need one AIO comes with one. And if you need to use your own, just create an user with the name oc_nextcloud, that should work

**Reply by Proxymiity** [reply]
Wouldn't that error out with the different grants as they evolve?

**Reply by Zoey2936** [reply]
> > that is not supported, please follow the Readme
> 
> That's stupid, please follow your users needs.

"Nextcloud AIO stands for Nextcloud All-in-One and provides easy deployment and maintenance with most features included in this one Nextcloud instance." — if you want it unnecessarily complicated, just use the official nextcloud docker image. I don't understand why AIO should support external databases...

**Reply by Zoey2936** [reply, marked as answer]
> Wouldn't that error out with the different grants as they evolve?

what do you mean? AIO uses the default username oc_nextcloud, so if you think it works with the username hello_nextcloud, then it should also work if you use oc_nextcloud as username inside your own database server

**Reply by mben42** [reply]
> > > that is not supported, please follow the Readme
> > 
> > 
> > That's stupid, please follow your users needs.
> 
> "Nextcloud AIO stands for Nextcloud All-in-One and provides easy deployment and maintenance with most features included in this one Nextcloud instance." — if you want it unnecessarily complicated, just use the official nextcloud docker image. I don't understand why AIO should support external databases...

If you dont understand why users should have choice, stop making opensource development.

**Reply by Proxymiity** [reply]
> what do you mean? AIO uses the default username oc_nextcloud, so if you think it works with the username hello_nextcloud, then it should also work if you use oc_nextcloud as username inside your own database server

Sorry, I misread your previous comment. Yeah, that should do it, even if it's not the naming scheme I had planned to use. Thanks anyway :)

**Reply by Zoey2936** [reply]
> > > > that is not supported, please follow the Readme
> > > 
> > > 
> > > That's stupid, please follow your users needs.
> > 
> > 
> > "Nextcloud AIO stands for Nextcloud All-in-One and provides easy deployment and maintenance with most features included in this one Nextcloud instance." — if you want it unnecessarily complicated, just use the official nextcloud docker image. I don't understand why AIO should support external databases...
> 
> If you dont understand why users should have choice, stop make opensource development.

I don't talk for Nextcloud, I talk for my own... I do understand that people want choices, but not in this case, because I don't see a need and I think removing the oc_ will break existing instances, but as you said it is open source, you can create a PR. Also, maybe we have different opinions of open source, open source is from my view not, that someone wants a feature and gets it, it is not like you pay for something and have a right to get that feature if you want it.

**Reply by szaimen** [reply]
We will not add a possibility to adjust the oc prefix. So Zoey is right.

**Reply by jrstrunk** [reply]
> > > > that is not supported, please follow the Readme
> > > 
> > > 
> > > That's stupid, please follow your users needs.
> > 
> > 
> > "Nextcloud AIO stands for Nextcloud All-in-One and provides easy deployment and maintenance with most features included in this one Nextcloud instance." — if you want it unnecessarily complicated, just use the official nextcloud docker image. I don't understand why AIO should support external databases...
> 
> If you dont understand why users should have choice, stop making opensource development.

Lmao the choice you have with open source is to fork the project and change the code yourself, not to impose your wants on the maintainers. Thanks for all the good work devs.
